applications/main/nfc/helpers/nfc_supported_cards.c

```diff
@@ -155,9 +160,11 @@ static const NfcSupportedCardsPlugin* nfc_supported_cards_get_next_plugin(
         if(!furi_string_end_with_str(instance->file_path, NFC_SUPPORTED_CARDS_PLUGIN_SUFFIX))
             continue;
 
-        path_concat(NFC_SUPPORTED_CARDS_PLUGINS_PATH, instance->file_name, instance->file_path);
+        size_t trim_suffix =
+            furi_string_size(instance->file_path) - strlen(NFC_SUPPORTED_CARDS_PLUGIN_SUFFIX);
+        instance->file_name[trim_suffix] = '\0';
```

Comment:
Whole things looks like mess to me:
- Why name and path needed in NfcSupportedCardsLoadContext? 
- Why not to make path generation local?

Reply:
We don't need file path with in NfcSupportedCardsLoadContext in this approach.

I would prefer to allocate 256 bytes for file_name in heap instead of stack. NfcSupportedCardsLoadContext is freed after after plugin is loaded.